sympy/physics/control/lti.py

```diff
@@ -798,75 +931,125 @@ def doit(self, **kwargs):
             if res is None:
                 res = arg
             else:
-                num_ = res.num * arg.den + res.den * arg.num
-                den_ = res.den * arg.den
-                res = TransferFunction(num_, den_, self.var)
+                if self.is_SISO:
+                    if res.den == arg.den:
+                        num_, den_ = res.num + arg.num, res.den
+                    else:
+                        num_, den_ = res.num * arg.den + res.den * arg.num, res.den * arg.den
+                    res = TransferFunction(num_, den_, self.var)
+                else:
+                    if self.num_inputs == 1:
+                        a = [None] * self.num_outputs
+                        for x in range(self.num_outputs):
+                            a[x] = res.args[0][x] + arg.args[0][x]
+                    else:
+                        a = [[None] * self.num_inputs for _ in range(self.num_outputs)]
+                        for row in range(self.num_outputs):
+                            for col in range(self.num_inputs):
+                                a[row][col] = res.args[0][row][col] + arg.args[0][row][col]
+                    res = TransferFunctionMatrix(a, self.shape, self.var)
         return res
 
     def _eval_rewrite_as_TransferFunction(self, *args, **kwargs):
+        """ Parallel(tfm1, tfm2, Series(tfm3, tfm4, ...), ...) not allowed. """
+        if not self.is_SISO:
+            raise ValueError("Only transfer functions or a collection of transfer functions"
+                " is allowed in the arguments.")
+
+        return self.doit()
+
+    def _eval_rewrite_as_TransferFunctionMatrix(self, *args, **kwargs):
+        """ Parallel(tf1, tf2, Series(tf3, tf4, ...), ...) not allowed. """
+        if self.is_SISO:
+            raise ValueError("Only transfer function matrices or a collection of transfer function"
+                " matrices is allowed in the arguments.")
+
         return self.doit()
 
     def __add__(self, other):
-        if isinstance(other, (TransferFunction, Series)):
+        if isinstance(other, (TransferFunction, Series, TransferFunctionMatrix)):
+            if isinstance(other, TransferFunctionMatrix):
+                if self.is_SISO:
+                    raise ValueError("Only transfer function matrices are allowed in the "
+                        "parallel configuration.")
+            if isinstance(other, Series):
+                if ((self.is_SISO and not other.is_SISO) or (not self.is_SISO and other.is_SISO)):
```

Comment:
```suggestion
                if self.is_SISO != other.is_SISO:
```

Reply:
@namannimmo10 you left the instance here.